src/main/drivers/exti.h

```diff
@@ -24,6 +24,12 @@
 
 #include "drivers/io_types.h"
 
+typedef enum {
```

Comment:
Another possibility is to avoid indirection:
```suggestion
typedef enum {
#if defined(STM32F7)
    EXTI_TRIGGER_RISING = GPIO_MODE_IT_RISING,
    EXTI_TRIGGER_FALLING = GPIO_MODE_IT_FALLING,
    EXTI_TRIGGER_BOTH = GPIO_MODE_IT_RISING_FALLING
#elif defined(STM32F1) || defined(STM32F3) || defined(STM32F4)
    EXTI_TRIGGER_RISING = EXTI_Trigger_Rising,
    EXTI_TRIGGER_FALLING = EXTI_Trigger_Falling,
    EXTI_TRIGGER_BOTH = EXTI_Trigger_Rising_Falling
#else
# warning "Unknown CPU"
#endif
} extiTrigger_t;
```

But this will complicate supporting CLI configurable trigger type, so indirect approach may be a bit more future-proof ..

Reply:
I would keep the indirection ;)